src/utils/utils.ts

```diff
@@ -153,3 +156,119 @@ export const checkHasZipStructure = async (contentPath: string): Promise<boolean
+export const buildTextContent = (url: string, type: ItemType): string => {
+  if (type === ItemType.LINK) {
+    return `[InternetShortcut]\n${url}\n`;
+  }
+  return `[InternetShortcut]\n${url}\nAppURL=1\n`;
+};
+
+export const addItemToZip = async (args: {
+  item: Item;
+  archiveRootPath: string;
+  archive: Archiver;
+  member: Member;
+  fileServiceType: string;
+  iTM: ItemTaskManager;
+  runner: TaskRunner<Actor>;
+  fileTaskManager: FileTaskManager;
+  fileStorage: string;
+}) => {
+  const {
+    item,
+    archiveRootPath,
+    archive,
+    member,
+    fileServiceType,
+    iTM,
+    runner,
+    fileStorage,
+    fileTaskManager,
+  } = args;
+  // get item and its related data
+  const itemExtra = item.extra as Extra;
+
+  switch (item.type) {
+    case fileServiceType: {
+      let filepath = '';
+      let mimetype = '';
+      // check for service type and assign filepath, mimetype respectively
+      if (fileServiceType === FILE_ITEM_TYPES.S3) {
+        const s3Extra = item.extra as S3FileItemExtra;
+        ({ path: filepath, mimetype } = s3Extra.s3File);
+      } else {
+        const fileExtra = item.extra as LocalFileItemExtra;
+        ({ path: filepath, mimetype } = fileExtra.file);
+      }
+      // get file stream
+      const task = fileTaskManager.createDownloadFileTask(member, {
+        filepath,
+        itemId: item.id,
+        mimetype,
+        fileStorage,
+      });
+
+      const fileStream = (await runner.runSingle(task)) as ReadStream;
+      // build filename with extension if does not exist
+      let ext = path.extname(item.name);
+      if (!ext) {
+        // only add a dot in case of building file name with mimetype, otherwise there will be two dots in file name
+        ext = `.${mime.extension(mimetype)}`;
+      }
+      // remove . here to avoid double . in filename
+      const filename = `${path.basename(item.name, ext)}${ext}`;
+
+      // add file in archive
+      archive.append(fileStream, {
+        name: path.join(archiveRootPath, filename),
+      });
+
+      break;
+    }
+    case ItemType.DOCUMENT:
+      archive.append(itemExtra.document?.content, {
+        name: path.join(archiveRootPath, `${item.name}.graasp`),
+      });
+      break;
+    case ItemType.LINK:
+      archive.append(buildTextContent(itemExtra.embeddedLink?.url, ItemType.LINK), {
+        name: path.join(archiveRootPath, `${item.name}.url`),
+      });
+      break;
+    case ItemType.APP:
+      archive.append(buildTextContent(itemExtra.app?.url, ItemType.APP), {
+        name: path.join(archiveRootPath, `${item.name}.url`),
+      });
+      break;
+    case ItemType.FOLDER: {
+      // append description
+      const folderPath = path.join(archiveRootPath, item.name);
+      if (item.description) {
+        archive.append(item.description, {
+          name: path.join(folderPath, `${item.name}.description.html`),
+        });
+      }
+      // eslint-disable-next-line no-case-declarations
```

Comment:
remove?

Reply:
This line is necessary to prevent eslint complain about the `const` declaration here